### PR TITLE
Adds ui for translation & pages in admin/content

### LINKED ISSUE #242: Add layout for pages & translation in admin/content (UI only)
**I'm submitting a ...**  (check one with "x")
- [ ] bug report
- [x] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-frontend

Pages:
screenshot from 2017-06-17 10-24-22

Translation:
screenshot from 2017-06-17 10-24-38

### PR DESCRIPTION
**Pull request by geekyd**
- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

Adds ui for translation & pages in admin/content. Adds functional ui for admin/content/pages along with a pages-form with validation.

Fixes #242 

Deployment Link: https://open-event-frontend-app.herokuapp.com

### CODE REVIEW

**Comment by utkarshnath** [comment]
@geekyd use `blue` buttons instead `teal` as this was made standard in #273

**Comment by geekyd** [comment]
@CosmicCoder96 @utkarshnath Made the requested changes please review again thanks.

**Comment by geekyd** [comment]
@niranjan94 Please review. Thanks

**Comment by geekyd** [comment]
@niranjan94 @CosmicCoder96 Removed conflicts please review again. Thanks

File: app/templates/components/forms/admin/content/pages-form.hbs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    <label>{{t 'Position'}}</label>
+    {{input type='number' name='position' value=data.position min=1}}
+  </div>
+  <div class="field">
+    <label>{{t 'Language'}}</label>
+    {{#each i18n.availableLocales as |language|}}
+      <div class="item" data-value="{{language.code}}">{{language.code}}</div>
+    {{/each}}
+    {{#ui-dropdown class='selection' selected=data.language}}
+      {{input type='hidden' name='language'}}
```
Comment (niranjan94) [inline app/templates/components/forms/admin/content/pages-form.hbs]: Bind `data.language` to this too

File: app/templates/components/forms/admin/content/pages-form.hbs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    {{input type='text' name='url' value=data.path}}
+  </div>
+  <div class="field">
+    <label>{{t 'Description'}}</label>
+    {{widgets/forms/rich-text-editor value=data.description}}
+  </div>
+  <div class="field">
+    <label>{{t 'Places'}}</label>
+    {{#ui-dropdown class='selection' selected=data.menu}}
+      {{input type='hidden' name='place'}}
```
Comment (niranjan94) [inline app/templates/components/forms/admin/content/pages-form.hbs]: Bind data.menu to this

File: app/templates/components/forms/admin/content/pages-form.hbs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      <i class="dropdown icon"></i>
+      <div class="default text">{{data.menu}}</div>
+      <div class="menu">
+        {{#each places as |place|}}
+          <div class="item  {{if (eq place data.menu) 'active selected'}}" data-value="{{place}}">{{place}}</div>
```
Comment (niranjan94) [inline app/templates/components/forms/admin/content/pages-form.hbs]: Current active will be taken care of by `ui-dropdown` itself

File: app/routes/admin/content/pages.js [hunk 4]
```diff
@@ -4,6 +4,36 @@ const { Route } = Ember;
 
 export default Route.extend({
   titleToken() {
-    return this.l10n.t('Pages');
+    return this.i10n.t('Pages');
```
Comment (CosmicCoder96) [inline app/routes/admin/content/pages.js]: @geekyd it's not `i10n` it's `l10n`

File: app/routes/admin/content/translations.js [hunk 5]
```diff
@@ -1,9 +1,17 @@
 import Ember from 'ember';
-
 const { Route } = Ember;
 
 export default Route.extend({
   titleToken() {
-    return this.l10n.t('Translations');
+    return this.i10n.t('Translations');
```
Comment (CosmicCoder96) [inline app/routes/admin/content/translations.js]: it's not `i10n` it's `l10n`. This change is not required.

File: app/components/forms/admin/content/pages-form.js [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+      inline : true,
+      delay  : false,
+      on     : 'blur',
+      fields : {
+        name: {
+          identifier : 'name',
+          rules      : [
+            {
+              type   : 'empty',
+              prompt : this.i18n.t('Please enter a name')
```
Comment (CosmicCoder96) [inline app/components/forms/admin/content/pages-form.js]: This is not in accordance with the new localisation standard, please change it to `l10n`

File: app/components/forms/admin/content/pages-form.js [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+            }
+          ]
+        },
+        title: {
+          identifier : 'title',
+          rules      : [
+            {
+              type   : 'empty',
+              prompt : this.i18n.t('Please enter a title')
```
Comment (CosmicCoder96) [inline app/components/forms/admin/content/pages-form.js]: This is not in accordance with the new localisation standard, please change it to `l10n`

File: app/components/forms/admin/content/pages-form.js [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+            }
+          ]
+        },
+        url: {
+          identifier : 'url',
+          rules      : [
+            {
+              type   : 'empty',
+              prompt : this.i18n.t('Please enter an url')
```
Comment (CosmicCoder96) [inline app/components/forms/admin/content/pages-form.js]: This is not in accordance with the new localisation standard, please change it to `l10n`

File: app/components/forms/admin/content/pages-form.js [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+            }
+          ]
+        },
+        place: {
+          identifier : 'place',
+          rules      : [
+            {
+              type   : 'empty',
+              prompt : this.i18n.t('Please select a place')
```
Comment (CosmicCoder96) [inline app/components/forms/admin/content/pages-form.js]: This is not in accordance with the new localisation standard, please change it to `l10n`

File: app/components/forms/admin/content/pages-form.js [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+            }
+          ]
+        },
+        position: {
+          identifier : 'position',
+          rules      : [
+            {
+              type   : 'empty',
+              prompt : this.i18n.t('Please enter a position')
```
Comment (CosmicCoder96) [inline app/components/forms/admin/content/pages-form.js]: This is not in accordance with the new localisation standard, please change it to `l10n`

File: app/components/forms/admin/content/pages-form.js [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+            }
+          ]
+        },
+        language: {
+          identifier : 'language',
+          rules      : [
+            {
+              type   : 'empty',
+              prompt : this.i18n.t('Please enter a language')
```
Comment (CosmicCoder96) [inline app/components/forms/admin/content/pages-form.js]: This is not in accordance with the new localisation standard, please change it to `l10n`

File: app/components/forms/admin/content/pages-form.js (hunk 6 above)
Comment (geekyd) [inline app/components/forms/admin/content/pages-form.js]: Totally missed it. Updated please check now.

**Comment by geekyd** [comment]
@niranjan94 @CosmicCoder96 Update. Please review.

**Comment by geekyd** [comment]
@niranjan94 I have rebased & pushed the commits, please review. Thanks

File: app/templates/admin/content/translations.hbs [hunk 12]
```diff
@@ -1 +1,50 @@
...
+            <h4 class="ui header">{{t 'Language'}}</h4>
+          </div>
+          <div class="column eight wide">
+            <h4 class="ui header">{{t 'Download .PO files'}}</h4>
+          </div>
+        </div>
+        {{#each model as |language|}}
+          <div class="row">
+            <div class="column eight wide">
+              {{get i18n.availableLocales language.code}}
```
Comment (CosmicCoder96) [inline app/templates/admin/content/translations.hbs]: This should be `{{get l10n.availableLocales language.code}}` ?

File: app/templates/admin/content/translations.hbs [hunk 13]
```diff
@@ -1 +1,50 @@
...
+            {{input type='file'}}
+          </div>
+          <div class="field {{if device.isMobile 'sixteen' 'eight'}} eight wide">
+            {{#ui-dropdown class="selection"}}
+              <div class="default text">{{t 'Language'}}</div>
+              <i class="dropdown icon"></i>
+              <div class="menu">
+                {{#each model as |language|}}
+                  <div class="item" data-value="{{language.code}}">
+                    {{get i18n.availableLocales language.code}}
```
Comment (CosmicCoder96) [inline app/templates/admin/content/translations.hbs]: This should be `{{get l10n.availableLocales language.code}}` ?

File: app/templates/admin/content/translations.hbs (hunk 12 above)
Comment (geekyd) [inline app/templates/admin/content/translations.hbs]: Thanks for pointing it out. I have made the changes. :)

**Comment by geekyd** [comment]
@niranjan94 @CosmicCoder96 Made the requested changes please review again. Thanks

**Comment by geekyd** [comment]
@CosmicCoder96 Thanks for pointing it out. I have made some changes please review. Thanks

File: app/styles/partials/utils.scss [hunk 14]
```diff
@@ -1,3 +1,7 @@
 .x-scrollable {
   overflow-x: auto;
 }
+
+.selection.dropdown.no-min-width {
+  min-width: 0em;
```
Comment (geekyd) [inline app/styles/partials/utils.scss]: @niranjan94 @CosmicCoder96 
`ui-dropdown` has a min-width of 14em which overflows when wrapped around a fixed width element. In this case field.

**Comment by geekyd** [comment]
@niranjan94 @CosmicCoder96 Please review. Thanks

File: app/styles/partials/utils.scss (hunk 14 above)
Comment (niranjan94) [inline app/styles/partials/utils.scss]: Can't you make it fluid and let it take the full width ?

Comment (geekyd) [inline app/styles/partials/utils.scss]: Ahhh yes. Thanks for pointing it out

File: app/templates/admin/content/translations.hbs [hunk 15]
```diff
@@ -1 +1,48 @@
...
+      </div>
+    </div>
+    <div class="column eight wide center aligned ui grid">
+      <h4 class="ui header">{{t 'Update .PO files :'}}</h4>
+      <form class="ui form column sixteen wide">
+        <div class="{{if device.isMobile 'field' 'two fields'}}">
+          <div class="field">
+            {{input type='file'}}
+          </div>
+          {{#ui-dropdown class='selection fluid field'}}
```
Comment (CosmicCoder96) [inline app/templates/admin/content/translations.hbs]: Please add form validation here, it's there on orga server as well.

File: app/templates/admin/content/translations.hbs [hunk 16]
```diff
@@ -1 +1,48 @@
...
+          </div>
+        {{/each}}
+      </div>
+    </div>
+    <div class="column eight wide center aligned ui grid">
+      <h4 class="ui header">{{t 'Update .PO files :'}}</h4>
+      <form class="ui form column sixteen wide">
+        <div class="{{if device.isMobile 'field' 'two fields'}}">
+          <div class="field">
+            {{input type='file'}}
```
Comment (CosmicCoder96) [inline app/templates/admin/content/translations.hbs]: Please add validation and give it a name/id

**Comment by geekyd** [comment]
@CosmicCoder96 
The buttons are of uneven height because on increasing the height of the dropdown the height abruptly increases when it fails validation.
screenshot from 2017-06-23 00-31-38

@fossasia/open-event-frontend Please review. Thanks

**Comment by CosmicCoder96** [comment]
@geekyd I can't think of a way around it, one other option is, if we simply let go of this design and make it standard vertical two field layout What are your views @niranjan94 
Other than that, LGTM !

**Comment by geekyd** [comment]
@CosmicCoder96 Yes that will be good. @niranjan94 Please guide.

**Comment by niranjan94** [comment]
screen shot 2017-06-23 at 9 14 51 am

Am I missing something here ?

**Comment by geekyd** [comment]
@niranjan if you look closely the height of the file input & dropdown is different.

**Comment by geekyd** [comment]
@niranjan94 Please guide regarding the height of the field issue. Thanks

**Comment by niranjan94** [comment]
Issue is with the way semantic Ui styles the file input. Though i don't blame them. File inputs are always tricky to style. 

Lets go for a single column layout for now.

**Comment by geekyd** [comment]
@niranjan94 I'm sorry, I did not get it. Do you mean having then in a same row or having an individual row for both?

**Comment by niranjan94** [comment]
Single column layout. One below the other.

**Comment by geekyd** [comment]
@CosmicCoder96 @niranjan94 Made the changes please review.
https://open-event-frontend-app2.herokuapp.com

**Comment by geekyd** [comment]
@niranjan94 Please review & merge if there are no pending changes. Thanks

**Pull request merged**